workflow-ui/compose/src/main/java/com/squareup/workflow1/ui/compose/ScreenComposableFactory.kt

```diff
@@ -0,0 +1,233 @@
+package com.squareup.workflow1.ui.compose
+
+import android.content.Context
+import android.view.ViewGroup
+import androidx.activity.OnBackPressedDispatcherOwner
+import androidx.activity.compose.LocalOnBackPressedDispatcherOwner
+import androidx.activity.setViewTreeOnBackPressedDispatcherOwner
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.platform.ComposeView
+import androidx.compose.ui.platform.LocalLifecycleOwner
+import androidx.compose.ui.viewinterop.AndroidView
+import androidx.lifecycle.setViewTreeLifecycleOwner
+import com.squareup.workflow1.ui.Screen
+import com.squareup.workflow1.ui.ScreenViewFactory
+import com.squareup.workflow1.ui.ScreenViewHolder
+import com.squareup.workflow1.ui.ViewEnvironment
+import com.squareup.workflow1.ui.ViewRegistry
+import com.squareup.workflow1.ui.ViewRegistry.Key
+import com.squareup.workflow1.ui.WorkflowUiExperimentalApi
+import com.squareup.workflow1.ui.androidx.OnBackPressedDispatcherOwnerKey
+import com.squareup.workflow1.ui.show
+import com.squareup.workflow1.ui.startShowing
+import kotlin.reflect.KClass
+
+@WorkflowUiExperimentalApi
+public inline fun <reified ScreenT : Screen> ScreenComposableFactory(
```

Comment:
trying to understand this better, i was thinking this would of been of type `ComposeScreen` to make it more assertive to only be used with compose; but after reading the comment below `Use [ScreenComposableFactory] directly if you need to prevent your [Screen] rendering classes from depending on Compose at compile time.` and the being able to inject things like `NamedScreen`, `EnvironmentScreen` into the compose content - is the reasoning why?

maybe having that sort of comment near the top why it isn't `ComposeScreen` would be helpful for readability.

Reply:
`ComposeScreen` is a possible implementation, one of many possible concrete implementations of `Screen`. Tying this to `ComposeScreen` would be like tying it to `ArrayList` instead of `List`. Really, `ComposeScreen` is just a convenience so that you don't have to deal with `ScreenComposableFactory` directly.

Reply:
I'll find a way to get that across better in the docs.

Reply:
@ekeitho I just pushed up a separate SHA that tweaks some kdoc, PTAL.

Reply:
this makes a lot more sense to me, thank you!